wgpu-hal/src/gles/wgl.rs

```diff
@@ -0,0 +1,750 @@
+            String::from("unable to create memory device"),
+            Error::last_os_error(),
+        ));
+    }
+    unsafe { setup_pixel_format(dc)? };
+
+    // We intentionally leak the window class, window and device context handle to avoid
+    // spawning a thread to destroy them. We cannot use `DestroyWindow` and `ReleaseDC` on
+    // different threads.
+
+    Ok(dc)
+}
+
+fn get_global_device_context() -> Result<HDC, crate::InstanceError> {
+    #[derive(Clone, Copy)]
+    struct SendDc(HDC);
+    unsafe impl Sync for SendDc {}
+    unsafe impl Send for SendDc {}
+
+    static GLOBAL: Mutex<Option<Result<SendDc, crate::InstanceError>>> = Mutex::new(None);
+    let mut guard = GLOBAL.lock();
+    if guard.is_none() {
+        *guard = Some(create_global_device_context().map(SendDc));
+    }
+    guard.clone().unwrap().map(|dc| dc.0)
+}
+
+impl crate::Instance<super::Api> for Instance {
+    unsafe fn init(desc: &crate::InstanceDescriptor) -> Result<Self, crate::InstanceError> {
+        let opengl_module = unsafe { LoadLibraryA("opengl32.dll\0".as_ptr() as *const _) };
+        if opengl_module.is_null() {
+            return Err(crate::InstanceError::with_source(
+                String::from("unable to load the OpenGL library"),
+                Error::last_os_error(),
+            ));
+        }
+
+        let dc = get_global_device_context()?;
+
+        let context = unsafe { wglCreateContext(dc) };
+        if context.is_null() {
+            return Err(crate::InstanceError::with_source(
+                String::from("unable to create initial OpenGL context"),
+                Error::last_os_error(),
+            ));
+        }
+        let context = WglContext {
+            context,
+            device: dc,
+        };
+        context.make_current().map_err(|e| {
+            crate::InstanceError::with_source(
+                String::from("unable to set initial OpenGL context as current"),
+                e,
+            )
+        })?;
+
+        let extra = Wgl::load_with(|name| load_gl_func(name, None));
+        let extentions = extensions(&extra, dc);
+
+        let can_use_profile = extentions.contains("WGL_ARB_create_context_profile")
+            && extra.CreateContextAttribsARB.is_loaded();
+
+        let context = if can_use_profile {
+            let context = unsafe {
+                extra.CreateContextAttribsARB(
+                    dc as *const _,
+                    ptr::null(),
+                    [
+                        CONTEXT_PROFILE_MASK_ARB as c_int,
+                        CONTEXT_CORE_PROFILE_BIT_ARB as c_int,
+                        CONTEXT_FLAGS_ARB as c_int,
+                        if desc.flags.contains(InstanceFlags::DEBUG) {
+                            CONTEXT_DEBUG_BIT_ARB as c_int
+                        } else {
+                            0
+                        },
+                        0, // End of list
+                    ]
```

Comment:
I know this is correct, but this is super spooky from a lifetime perspective, could you lift this array out to a binding to make it very clear it lives long enough.